rbs/CommentsAssociator.cc

```diff
@@ -0,0 +1,282 @@
+#include "rbs/CommentsAssociator.h"
+
+#include "absl/strings/match.h"
+#include "absl/strings/str_split.h"
+#include "common/typecase.h"
+#include "core/errors/rewriter.h"
+#include "parser/helper.h"
+#include "rbs/SignatureTranslator.h"
+
+using namespace std;
+
+namespace sorbet::rbs {
+
+const std::string_view CommentsAssociator::RBS_PREFIX = "#:";
+const std::string_view CommentsAssociator::ANNOTATION_PREFIX = "# @";
+const std::string_view CommentsAssociator::MULTILINE_RBS_PREFIX = "#|";
+
+namespace {
+bool isVisibilitySend(const parser::Send *send) {
+    return send->receiver == nullptr && send->args.size() == 1 &&
+           (parser::isa_node<parser::DefMethod>(send->args[0].get()) ||
+            parser::isa_node<parser::DefS>(send->args[0].get())) &&
+           (send->method == core::Names::private_() || send->method == core::Names::protected_() ||
+            send->method == core::Names::public_() || send->method == core::Names::privateClassMethod() ||
+            send->method == core::Names::publicClassMethod() || send->method == core::Names::packagePrivate() ||
+            send->method == core::Names::packagePrivateClassMethod());
+}
+
+bool isAttrAccessorSend(const parser::Send *send) {
+    return (send->receiver == nullptr || parser::isa_node<parser::Self>(send->receiver.get())) &&
+           (send->method == core::Names::attrReader() || send->method == core::Names::attrWriter() ||
+            send->method == core::Names::attrAccessor());
+}
+}; // namespace
+
+void CommentsAssociator::consumeCommentsUntilLine(int line) {
+    for (auto it = commentByLine.begin(); it != commentByLine.end();) {
+        if (it->first < line) {
+            if (absl::StartsWith(it->second.string, RBS_PREFIX) ||
+                absl::StartsWith(it->second.string, MULTILINE_RBS_PREFIX)) {
+                if (auto e = ctx.beginError(it->second.loc, core::errors::Rewriter::RBSUnusedComment)) {
+                    e.setHeader("Unused RBS signature comment. No method definition found after it");
+                }
+            }
+            it = commentByLine.erase(it);
+        } else {
+            break;
+        }
+    }
+}
+
+void CommentsAssociator::associateCommentsToLines() {
```

Comment:
I think this is only used in the constructor, what do you think about moving the constructor to this file, and inlining this function into it?